tests/unit/s2n_policy_writer_test.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <fcntl.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/policy/s2n_policy_feature.h"
+
+/* A custom cleanup function for unlink 
+ * We need this because unlink expects a const char* but DEFER_CLEANUP passes
+ * the address of the variable, so we need a function that takes a char** and
+ * calls unlink with the dereferenced pointer.
+ */
+static void s2n_unlink_cleanup(char **filename)
+{
+    if (filename && *filename) {
+        unlink(*filename);
+    }
+}
```

Comment:
Can you use https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety.h#L89-L98?